### PR TITLE
Argparse intitial commit.

### PR DESCRIPTION
**Pull request by afif-ishamsyah**
1. Argument for mask added
2. Create argument to choose either python or CUDA
2. implement assertion for mode and type

### CODE REVIEW

File: cuda_phase-retrieval/run.py [hunk 1]
```diff
@@ -9,45 +9,49 @@
 
 parser = argparse.ArgumentParser()
 parser.add_argument("--image", help="Image path")
-parser.add_argument("--step", help="number of iterations (default 20)", type=int, default=20)
-parser.add_argument("--beta", help="beta (default 0.8)", type=float, default=0.8)
-parser.add_argument("--type", help="Hybrid, Input-Ouput, or Output-Output", default="hybrid")
+parser.add_argument("--mask", help="Mask value (default 1). This will automatically create array of the specified value with the same size of the input image", type=float, default=1)
```
Comment (SimeonEhrig) [inline cuda_phase-retrieval/run.py:12]: Please overwork the help sentence. It is hard to understand. Maybe:

"Initial mask value (default value 1.0). The mask is created automatically and has the size of the input image. All values of the mask are initialized with this value."

File: cuda_phase-retrieval/run.py [hunk 2]
```diff
@@ -9,45 +9,49 @@
...
+parser.add_argument("--step", help="Number of iterations (default 20)", type=int, default=20)
+parser.add_argument("--beta", help="Beta (default 0.8)", type=float, default=0.8)
+parser.add_argument("--mode", help="Hybrid, Input-Ouput, or Output-Output", default="hybrid")
```
Comment (SimeonEhrig) [inline cuda_phase-retrieval/run.py:15]: `argparse` provides the option `choices` to restrict the input options: https://github.com/ComputationalRadiationPhysics/xeus-cling-cuda-container/blob/7138519646ee9d2918bf3bb006830cfdec981889/dev_container.py#L31
It generates automatically the help string and controls the argument parameter. You can use it to replace your manual check.

File: cuda_phase-retrieval/run.py [hunk 3]
```diff
@@ -9,45 +9,49 @@
...
+parser.add_argument("--type", help="Language used to run phase retrieval. CUDA (default) or Python. ", default="cuda")
```
Comment (SimeonEhrig) [inline cuda_phase-retrieval/run.py:16]: Mode `Python` does not work:

```bash
$ python run.py --image example_images/a.png  --type Python
Running phase retrieval...
Traceback (most recent call last):
  File "run.py", line 41, in <module>
    result = phase_retrieval_python.fienup_phase_retrieval(image, mask, mode, type, beta, array_random)
  File "/home/sehrig/project/student_project_python_bindings/cuda_phase-retrieval/build/phase_retrieval_python.py", line 37, in fienup_phase_retrieval
    assert steps > 0, 'steps must be a positive number'
TypeError: '>' not supported between instances of 'str' and 'int'
```

**Review by SimeonEhrig — CHANGES_REQUESTED** [review]

**Comment by SimeonEhrig** [comment]
Really useful work. I played a little bit around with different parameters and found out, that the mathematical algorithm is pretty instable for other values :sweat_smile:

File: cuda_phase-retrieval/run.py (hunk 3 above)
Comment (afif-ishamsyah) [reply cuda_phase-retrieval/run.py:16]: yes I forgot to change the parameters in the python part

**Review by SimeonEhrig — APPROVED** [review]

**Pull request merged**